File: app/src/main/java/com/wkz/viewpager/transformer/ScaleInOutTransformer.java
package com.wkz.viewpager.transformer;

import android.view.View;

import com.nineoldandroids.view.ViewHelper;

/**
 * @desc 向内收缩向外扩展的转换效果
 * @date 2017-06-30
 */
public class ScaleInOutTransformer extends AbstractBaseTransformer {

    @Override
    protected void onTransform(View page, float position) {
        ViewHelper.setPivotX(page, position < 0 ? 0 : page.getWidth());
        ViewHelper.setPivotY(page, page.getHeight() / 2f);
        float scale = position < 0 ? 1f + position : 1f - position;
        ViewHelper.setScaleX(page, scale);
        ViewHelper.setScaleY(page, scale);
    }
}

File: view-pager-transformer/src/main/java/com/fphoenixcorneae/transformer/ScaleInOutTransformer.kt
package com.fphoenixcorneae.transformer

import android.view.View

/**
 * @desc 向内收缩向外扩展的转换效果
 * @date 2017-06-30
 */
class ScaleInOutTransformer : AbstractBaseTransformer() {

    override fun onTransform(page: View, position: Float) {
        page.apply {
            pivotX = if (position < 0) {
                0f
            } else {
                width.toFloat()
            }
            pivotY = height / 2f
            val scale = if (position < 0) {
                1f + position
            } else {
                1f - position
            }
            scaleX = scale
            scaleY = scale
        }
    }
}